afs-core/src/main/java/com/powsybl/afs/AbstractNodeBase.java

```diff
@@ -175,16 +185,47 @@ private boolean nodeNameAlreadyExists(String name) {
         return childNodes.stream().filter(nodeInfo -> !nodeInfo.getId().equals(getId())).anyMatch(nodeInfo -> nodeInfo.getName().equals(name));
     }
 
-    public void archive(Path dir) {
+    public void archive(Path dir, boolean useZip){
+
+        //if it's a copy paste, this test is already done
+        if(useZip) {
+            File archiveFile = dir.toFile();
```

Comment:
the check on disk space shouldn't be embeded in archive method.
this is an opiniated check that user (dev) should do in user code elsewhere.